README.md

```diff
@@ -162,10 +162,14 @@ PLUGINS_CONFIG = {
 ```
 
 - `url` - (required) The URL to the HashiCorp Vault instance (e.g. `http://localhost:8200`).
-- `auth_method` - (optional / defaults to "token") The method used to authenticate against the HashiCorp Vault instance. Either `"token"` or `"approle"`.
+- `auth_method` - (optional / defaults to "token") The method used to authenticate against the HashiCorp Vault instance. Either `"approle"`, `"aws"`, `"kubernetes"` or `"token"`.
+- `ca_cert` - (optional) Path to a PEM formatted CA certificatee to use when verifying the vault connection.  Set to `False` to ignore ssl verification (not recommended) or `True` to use the system certificates.
+- `k8s_token_path` - (optional) Path to the kubernetes service account token file.  Defaults to "/var/run/secrets/kubernetes.io/serviceaccount/token".
 - `token` - (optional) Required when `"auth_method": "token"` or `auth_method` is not supplied. The token for authenticating the client with the HashiCorp Vault instance. As with other sensitive service credentials, we recommend that you provide the token value as an environment variable and retrieve it with `{"token": os.getenv("NAUTOBOT_HASHICORP_VAULT_TOKEN")}` rather than hard-coding it in your `nautobot_config.py`.
+- `role_name` - (optional) Required when `"auth_method": "kubernetes"`.  The Vault Kubernetes role to assume which the pod's service account has access to.
```

Comment:
Would `k8s_role_name` be at all preferable to make this less ambiguous with `role_id`?

(Also, I see in the code below that we use `role_name` also for `aws` auth_method -- does the above need to be updated?

Reply:
Within Vault there are role names for k8s and for was.  There are role ids and secret ids only for the auth method of "AppRole".  I do not know why AWS authentication IAM login does not require `role_name` but according to the [docs](https://hvac.readthedocs.io/en/stable/source/hvac_api_auth_methods.html#hvac.api.auth_methods.Aws.iam_login) it does not.  I prefer to leave `role_id` and `role_name` as they are as they have contextual meaning with in Hashicorp Vault.

Reply:
So perhaps:

```suggestion
- `role_name` - (optional) Required when `"auth_method": "kubernetes"`, optional when `"auth_method": "aws"`.  The Vault Kubernetes role or AWS role to assume which the pod's service account has access to.
```

Reply:
Changed to `The Vault Kubernetes role or Vault AWS role` as I don't want these to be confused with either Kubernetes cluster roles or AWS IAM roles.  Term overloaded successfully